src/plugins/gs-plugin-eos.c

```diff
@@ -264,6 +265,155 @@ app_is_renamed (GsApp *app)
 			  "eos-desktop") == 0;
 }
 
+/* Copied from the private as_locale_to_language that is part of
+ * appstream-glib's as-utils.c */
+static gchar *
+locale_to_language (const gchar *locale)
+{
+	gchar *tmp;
+	gchar *country_code;
+
+	/* invalid */
+	if (locale == NULL)
+		return NULL;
+
+	/* return the part before the _ (not always 2 chars!) */
+	country_code = g_strdup (locale);
+	tmp = g_strstr_len (country_code, -1, "_");
+	if (tmp != NULL)
+		*tmp = '\0';
+	return country_code;
+}
+
+static gboolean
+gs_plugin_locale_is_compatible (GsPlugin *plugin,
+				const char *locale)
+{
+	g_autofree char *lang = NULL;
+	const char *plugin_locale = gs_plugin_get_locale (plugin);
+	const char *plugin_lang = gs_plugin_get_language (plugin);
+
+	/* if we don't know the current locale than all locales should be
+	 * compatible */
+	if (!plugin_locale)
+		return TRUE;
+
```

Comment:
I think what I would do here is
- call `g_get_locale_variants(locale)`
- for each of the strings returned there, compare them with both `plugin_locale` and `plugin_lang`, returning `TRUE` if either match

This way we don't need to copy `locale_to_language()`, and the code should be a bit simpler to follow